### PR TITLE
[Merged by Bors] - refactor: move slim_check to plausible

### PR DESCRIPTION
**Pull request by hargoniX**
This moves Mathlib to the now pulled out (and already slightly improved)
version of slim_check at https://github.com/leanprover-community/plausible/.
I removed both all instances and logic as well as all tests that are already
covered by plausible itself and only left what Mathlib needs for itself.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: test/slim_check.lean
Comment (eric-wieser) [inline test/slim_check.lean:1]: Can we keep this rest to prove in this PR that the behavior is unchanged? We can delete the test shortly after this PR is merged, but its presence would be useful for review.

Comment (hargoniX) [reply test/slim_check.lean:1]: The test moved to tests/plausible, the other tests are in https://github.com/leanprover-community/plausible/blob/main/Test/Tactic.lean. What tests specifically are you looking for?

Comment (eric-wieser) [reply test/slim_check.lean:1]: I meant literally keeping this exact file, so that the diff in this PR shows any change in behavior.

**Comment by eric-wieser** [comment]
I think we should keep the `slim_check` tactic around in mathlib as a deprecated alias for `plausible`, so as to automatically correct the muscle memory of anyone using the old name.

File: test/slim_check.lean
Comment (eric-wieser) [reply test/slim_check.lean:1]: Or at least a middle ground; do not rename it to `plausible.lean` in this PR, to make it clear which tests were deleted.

Comment (hargoniX) [reply test/slim_check.lean:1]: Done.

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**